instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -118,12 +122,12 @@ def _uninstrument(self, **kwargs):
         unwrap(Endpoint, "prepare_request")
 
     @staticmethod
-    def _is_lambda_invoke(service_name, operation_name, api_params):
+    def _is_lambda_invoke(call_context: _AwsSdkCallContext):
         return (
-            service_name == "lambda"
-            and operation_name == "Invoke"
-            and isinstance(api_params, dict)
-            and "Payload" in api_params
+            call_context.service == "lambda"
+            and call_context.operation == "Invoke"
+            and isinstance(call_context.params, dict)
+            and "Payload" in call_context.params
```

Comment:
curious if there is a case where service is "lambda", operator is "Invoke" but params is not a dict or "Payload" is not present in params. Can that happen? If it can, why don't we want to consider it a "lambda invoke"? If it can't, why check for params at all?

Reply:
I think the name of the method is misleading as it actually checks if it is a lambda invoke operation and if the trace context can be injected into the `Payload` (the code already existed before so not sure).

From how botocore is implemented the `params` object should always be a dictionary. It contains all the parameters that are passed to the boto3/botocore API method. So if no `Payload` argument is passed to `lambda.invoke(...)` it will not show up in the `params` object.

The idea is to move the lambda-specific tracing part into a separate extension like proposed in the [initial PR](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/634/files#diff-abc82a132764a6244c38e8e72e4b15c966c275c02aa13fdae8327c87ee5fe1ddR65) to extract the additional attributes that are defined in the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/faas.md#outgoing-invocations).